### PR TITLE
Adding a sample for the OnnxCatalog.DnnFeaturizeImage extension

### PR DESCRIPTION
**Pull request by sfilipi**
Adding to #1209 an example for OnnxCatalog.DnnFeaturizeImage

### CODE REVIEW

File: docs/samples/Microsoft.ML.Samples/Dynamic/ImageAnalytics/DnnFeaturizeImage.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                        new TextLoader.Column("ImagePath", DataKind.String, 0),
+                        new TextLoader.Column("Name", DataKind.String, 1),
+                }
+            }).Load(imagesDataFile);
+
+            // Installing the Microsoft.ML.DNNImageFeaturizer packages copies the models in the
+            // `DnnImageModels` folder. The location of this folder is needed by the ResNet18 model.
+            string executablePath = typeof(DnnFeaturizeImage).Assembly.CodeBase;
+            string absolutePath = new Uri(executablePath).AbsolutePath;
+            var modelsParentFolder = Path.Combine(Directory.GetParent(absolutePath).FullName, "DnnImageModels");
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/ImageAnalytics/DnnFeaturizeImage.cs]: wow, does the user really have to do this? Can we have an API for it instead?

File: src/Microsoft.ML.OnnxTransformer/OnnxCatalog.cs [hunk 2]
```diff
@@ -63,14 +63,23 @@ public static class OnnxCatalog
...
         /// <param name="modelFactory">An extension method on the <see cref="DnnImageModelSelector"/> that creates a chain of two
         /// <see cref="OnnxScoringEstimator"/> (one for preprocessing and one with a pretrained image DNN) with specific models
         /// included in a package together with that extension method.</param>
-        /// <param name="inputColumnName">Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
+        /// <param name="inputColumnName">Name of column to transform.
+        /// If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[StochasticGradientDescent](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/ImageAnalytics/DnnFeaturizeImage.cs)]
```
Comment (shmoradims) [inline src/Microsoft.ML.OnnxTransformer/OnnxCatalog.cs]: >ImageAnalytics [](start = 113, length = 14)

I think ImageAnalytics should go under Transforms. The hierarchy under Microsoft.ML.Samples/Dynamic should mirror the catalog hierarchy as much as possible.

File: docs/samples/Microsoft.ML.Samples/Dynamic/ImageAnalytics/DnnFeaturizeImage.cs (hunk 1 above)
Comment (sfilipi) [reply docs/samples/Microsoft.ML.Samples/Dynamic/ImageAnalytics/DnnFeaturizeImage.cs]: it is due to bug: [https://github.com/dotnet/machinelearning/issues/2785](https://github.com/dotnet/machinelearning/issues/2785)

Once that is fixed, the users should not have to specify the folder, if they have already selected a model. 

i think we should leave it as is, until the bug gets fixed, than just remove the model path parameter. 

---
In reply to: [261430746](https://github.com/dotnet/machinelearning/pull/2799#discussion_r261430746) [](ancestors = 261430746)

Comment (Anipik) [reply docs/samples/Microsoft.ML.Samples/Dynamic/ImageAnalytics/DnnFeaturizeImage.cs]: https://github.com/dotnet/machinelearning/pull/2809 this should fix this.
I build the packages from the repo and consumed these packages as a local nuget feed in a netcoreapp2.1 console project

Comment (sfilipi) [reply docs/samples/Microsoft.ML.Samples/Dynamic/ImageAnalytics/DnnFeaturizeImage.cs]: I'll wait for that PR to get in than, edit this and merge it. 

---
In reply to: [261456118](https://github.com/dotnet/machinelearning/pull/2799#discussion_r261456118) [](ancestors = 261456118)

File: src/Microsoft.ML.OnnxTransformer/OnnxCatalog.cs (hunk 2 above)
Comment (shmoradims) [reply src/Microsoft.ML.OnnxTransformer/OnnxCatalog.cs]: thanks for moving the files. please fix this sample path and the path for the other transforms that got moved. 

---
In reply to: [261433488](https://github.com/dotnet/machinelearning/pull/2799#discussion_r261433488) [](ancestors = 261433488)

**Comment by zeahmed** [comment]
>             var mlContext = new MLContext();

Comments?

`// Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
// as a catalog of available operations and as the source of randomness.
// Setting the seed to a fixed number in this example to make outputs deterministic.`

---
Refers to: docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/ImageAnalytics/ConvertToGrayScale.cs:11 in 3124e0b. [](commit_id = 3124e0ba0be0e9870e9dc1d55842eda6812eaa13, deletion_comment = False)

File: docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/ImageAnalytics/DnnFeaturizeImage.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+using Microsoft.ML.Data;
+using Microsoft.ML.Transforms;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public static class DnnFeaturizeImage
+    {
+        public static void Example()
+        {
+            var mlContext = new MLContext();
```
Comment (zeahmed) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/ImageAnalytics/DnnFeaturizeImage.cs]: >var mlContext = new MLContext(); [](start = 12, length = 32)

comments?

File: src/Microsoft.ML.OnnxTransformer/OnnxCatalog.cs [hunk 4]
```diff
@@ -63,14 +63,23 @@ public static class OnnxCatalog
...
         /// <param name="modelFactory">An extension method on the <see cref="DnnImageModelSelector"/> that creates a chain of two
         /// <see cref="OnnxScoringEstimator"/> (one for preprocessing and one with a pretrained image DNN) with specific models
         /// included in a package together with that extension method.</param>
-        /// <param name="inputColumnName">Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
+        /// <param name="inputColumnName">Name of column to transform.
+        /// If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[StochasticGradientDescent](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/ImageAnalytics/DnnFeaturizeImage.cs)]
```
Comment (zeahmed) [inline src/Microsoft.ML.OnnxTransformer/OnnxCatalog.cs]: >StochasticGradientDescent [](start = 26, length = 25)

copy-paste???

**Comment by zeahmed** [comment]
>             var mlContext = new MLContext();

comments please.

---
Refers to: docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/ImageAnalytics/ExtractPixels.cs:12 in 3124e0b. [](commit_id = 3124e0ba0be0e9870e9dc1d55842eda6812eaa13, deletion_comment = False)

**Comment by zeahmed** [comment]
>             var mlContext = new MLContext();

comments please.

---
Refers to: docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/ImageAnalytics/LoadImages.cs:11 in 3124e0b. [](commit_id = 3124e0ba0be0e9870e9dc1d55842eda6812eaa13, deletion_comment = False)

**Comment by zeahmed** [comment]
>             var mlContext = new MLContext();

comments please.

---
Refers to: docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/ImageAnalytics/ResizeImages.cs:11 in 3124e0b. [](commit_id = 3124e0ba0be0e9870e9dc1d55842eda6812eaa13, deletion_comment = False)

**Pull request merged**